Samples/WinMLSamplesGallery/WinMLSamplesGallery/Common/File.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Runtime.InteropServices;
+using Windows.Storage.Pickers;
+using Windows.Storage;
+using Microsoft.UI.Xaml;
+using Windows.Graphics.Imaging;
+using Windows.Media;
+using Microsoft.UI.Xaml.Media.Imaging;
+
+namespace WinMLSamplesGallery.Common
+{
+    [ComImport, Guid("3E68D4BD-7135-4D10-8018-9FB6D9F33FA1"), InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+    public interface IInitializeWithWindow
+    {
+        void Initialize([In] IntPtr hwnd);
+    }
+
+    public static class File
+    {
+        [DllImport("user32.dll", ExactSpelling = true, CharSet = CharSet.Auto, PreserveSig = true, SetLastError = false)]
+        public static extern IntPtr GetActiveWindow();
+
+
+        public static StorageFile PickImageFiles()
+        {
+            FileOpenPicker openPicker = new FileOpenPicker();
+            openPicker.ViewMode = PickerViewMode.Thumbnail;
+            openPicker.FileTypeFilter.Add(".jpg");
+            openPicker.FileTypeFilter.Add(".bmp");
+            openPicker.FileTypeFilter.Add(".png");
+            openPicker.FileTypeFilter.Add(".jpeg");
+
+            // When running on win32, FileOpenPicker needs to know the top-level hwnd via IInitializeWithWindow::Initialize.
+            if (Window.Current == null)
+            {
+                var picker_unknown = Marshal.GetComInterfaceForObject(openPicker, typeof(IInitializeWithWindow));
+                var initializeWithWindowWrapper = (IInitializeWithWindow)Marshal.GetTypedObjectForIUnknown(picker_unknown, typeof(IInitializeWithWindow));
+                IntPtr hwnd = GetActiveWindow();
+                initializeWithWindowWrapper.Initialize(hwnd);
+            }
+
+            return openPicker.PickSingleFileAsync().GetAwaiter().GetResult();
+        }
+        public static BitmapDecoder CreateBitmapDecoderFromStorageFile(StorageFile file)
+        {
+            if (file == null)
+            {
+                return null;
+            }
+
+            var stream = file.OpenAsync(FileAccessMode.Read).GetAwaiter().GetResult();

```

Comment:
can we follow up on this "GetAwaiter()" pattern vs the "await" keyword? I don't understand when to use one or the other.

Reply:
I changed the batching sample to use the "await" keyword. @smk2007 preferred to use GetAwaiter instead but we can discuss which one works better